lib/commands/general.js

```diff
@@ -193,19 +193,21 @@ commands.getScreenHeight = async function () {
 };
 
 commands.getScreenInfo = async function () {
-  log.info("Getting screen info from WDA, and caching the result");
   return await this.proxyCommand('/wda/screen', 'GET');
 };
 
 commands.getStatusBarHeight = async function () {
   const info = await this.getScreenInfo();
-  return info.statusBarSize.height;
+  // xcui reeturns status bar size in logical pixels, not device pixels,
+  // so we need to scale up
+  return Math.round(info.statusBarSize.height * info.scale);
```

Comment:
I think it might be confusing to do this transformation here since all other values, like screen size and frames are in device pixels. I'd rather calculate the actual pixels inside `cropImage` helper